docs/pages/button.js

```diff
@@ -259,245 +229,481 @@ card(
-    id="selected"
-    defaultCode={`
-function Example() {
-  const [selected, setSelected] = React.useState(false);
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Icons"
+      description={`
+1. Icon end
+Adds an icon after the Button text. Icons should only be used to reinforce a specific function or interaction of the Button. Menus and external links are a common use case. Use \`arrow-up-right\` when linking to an external URL or \`arrow-down\` when displaying a Popover on click. Note that iconEnd on Button is not accessible to screen readers.
+`}
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+<Button
+  accessibilityLabel='Menu'
+  iconEnd="arrow-down"
+  size="lg"
+  text="Menu"
+/>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      columns={2}
+      title="Role"
+      description={`
+1. Button (default)
+    The “button” \`role\` is used for actions. This is the default and should be used for most Buttons.
+2. Link
+    The “link” \`role\` is used for navigating by URL. These Buttons should not use a \`selected\` state.
+`}
+    >
+      <MainSection.Card
+        cardSize="md"
+        defaultCode={`
+<Button
+  accessibilityLabel='Follow'
+  size="lg"
+  text="Follow"
+  role="button"
+/>
+`}
+      />
+      <MainSection.Card
+        cardSize="md"
+        defaultCode={`
+<Button
+  accessibilityLabel='Visit Pinterest'
+  iconEnd="arrow-up-right"
+  size="lg"
+  text="Visit Pinterest"
+  role="link"
+  href="https://pinterest.com"
+/>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="rel and target"
+      description={`
+These optional props control the behavior of \`role="link"\` Buttons. External links commonly use \`target="_blank"\` to open the link in a new tab or window, and \`rel="nofollow"\` to provide hints for SEO.
+`}
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+<Button
+  accessibilityLabel='Visit Pinterest'
+  iconEnd="arrow-up-right"
+  size="lg"
+  text="Visit Pinterest"
+  role="link"
+  rel="nofollow"
+  target="_blank"
+  href="https://pinterest.com"
+/>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      columns={2}
+      title="States"
+      description={`
+1. Default
+    The typical state of a Button that represents it can be interacted with and is not in a selected state.
+2. Disabled
+Used to block user interaction such as hover, focus and click. Disabled Buttons are completely unreachable by a keyboard and screenreader, so do not attach Tooltips to disabled Buttons. If you need to explain why a Button is disabled, use an IconButton + Tooltip next to the disabled Button.
+3. Selected
+  When Button is used to toggle a boolean state or control the visibility of other elements (e.g. Dropdown), use the \`selected\` prop to indicate the current state. Do not use this prop with \`role="link"\` Buttons.
```

Comment:
Should we include Tabs functionality? as per conversation here: 
https://pinterest.slack.com/archives/G01GN7CE99S/p1631231159047700

Reply:
I would suggest against this. The selected state is specifically for situations where you'r toggling states of a specific button (Follow/Following) (Save/Saved)

Reply:
OK, I added a Don't for this usecase, but given there's a style bug at the moment, I think we should probably fix this before publishing the new Button guidelines. Thoughts?